x-pack/legacy/plugins/siem/public/components/timeline/body/helpers.ts

```diff
@@ -95,6 +95,34 @@ export const getColumnHeaders = (
 };
 
 /** Returns the (fixed) width of the Actions column */
-export const getActionsColumnWidth = (isEventViewer: boolean, showCheckboxes = false): number =>
+export const getActionsColumnWidth = (
+  isEventViewer: boolean,
+  showCheckboxes = false,
+  additionalActionWidth = 0
+): number =>
   (showCheckboxes ? SHOW_CHECK_BOXES_COLUMN_WIDTH : 0) +
-  (isEventViewer ? EVENTS_VIEWER_ACTIONS_COLUMN_WIDTH : DEFAULT_ACTIONS_COLUMN_WIDTH);
+  (isEventViewer ? EVENTS_VIEWER_ACTIONS_COLUMN_WIDTH : DEFAULT_ACTIONS_COLUMN_WIDTH) +
+  additionalActionWidth;
+
+/**
+ * Creates mapping of eventID -> fieldData for given fieldsToKeep. Used to store additional field
+ * data necessary for custom timeline actions in conjunction with selection state
+ * @param timelineData
+ * @param eventIds
+ * @param fieldsToKeep
+ */
+export const getEventIdToDataMapping = (
```

Comment:
Let's make sure that we memoizing it

Reply:
The two functions calling this helper `onRowSelected`/`onSelectAll` within `stateful_body` are both memoized with `useCallback`, so this should be 👍 on the implementation front. Or are you thinking to wrap this helper in a memoize-one for anyone else using it?

Reply:
Yes we should be good